torch/_dynamo/eval_frame.py

```diff
@@ -343,6 +343,14 @@ def get_compiler_config():
 
         # add context containing GraphModule to any GraphModule forward functions
         if isinstance(fn, torch.fx.GraphModule):
+            # Since dynamo will run the forward method for the GraphModule shortly
+            # anyways, it does not hurt to do the real recompilation here if
+            # this is a _LazyGraphModule. This makes it easier for dynamo to
+            # optimize a _LazyGraphModule.
+            from torch.fx.lazy_graph_module import _LazyGraphModule
+
+            _LazyGraphModule.force_recompile(fn)
```

Comment:
This `force_recompile` does not introduce any overhead since we need call forward below anyways which will trigger the real recompilation. 

On the other hand, without this line, we will fail the test
```
PYTORCH_TEST_WITH_DYNAMO=1 python test/torch_np/numpy_tests/core/test_numeric.py -k test_boolean
```
on py3.11. It works on py3.10.

I don't know exactly why py3.11 can not handle it though.

Reply:
Should it be the case that accessing `gm.forward` triggers a recompile before you even call it. That would avoid this `force_recompile` and the one in inductor, but I'm not sure if it would introduce unnecessary compiles. wdyt?